src/screens/augmentedReality/ARShowScreen.js

```diff
@@ -14,21 +22,93 @@ export const ARShowScreen = ({ navigation, route }) => {
   }, []);
 
   const parser = async () => {
-    await objectParser({ item: data[index], setObject });
+    // TODO: the `onPress` prop is just for testing
+    await objectParser({ item: data[index], setObject, onPress: () => navigation.goBack() });
+
     setIsLoading(false);
   };
 
-  if (isLoading || !object) return <LoadingSpinner loading />;
+  if (isLoading || !object || !object.vrx) return <LoadingSpinner loading />;
 
   return (
-    <SafeAreaViewFlex>
-      <RegularText>AR Show Screen</RegularText>
-      <RegularText onPress={() => navigation.goBack()}>Zurück</RegularText>
-    </SafeAreaViewFlex>
+    <>
+      <ViroARSceneNavigator
+        autofocus
+        initialScene={{
+          scene: AugmentedRealityView
+        }}
+        viroAppProps={{ object }}
+        style={styles.arSceneNavigator}
+      />
+      <TouchableOpacity
+        style={styles.backButton}
+        onPress={() => {
+          /*
+          to solve the Android crash problem, you must first remove the 3D object from the screen. 
+          then navigation can be done.
+          */
+          setObject();
+          navigation.goBack();
+        }}
+      >
+        <Icon.Close color={colors.surface} />
+      </TouchableOpacity>
+    </>
   );
 };
 
-const objectParser = async ({ item, setObject }) => {
+const AugmentedRealityView = ({ sceneNavigator }) => {
+  const [rotation, setRotation] = useState([0, 0, 0]);
+  const [position, setPosition] = useState([0, -1, -5]);
+
+  const { object } = sceneNavigator.viroAppProps;
+
+  const moveObject = (newPosition) => {
+    setPosition(newPosition);
+  };
+
+  const rotateObject = (rotateState, rotationFactor) => {
+    let newRotation = [rotation[0], rotation[1] - rotationFactor, rotation[2]];
+
+    if (rotateState === 2) {
+      setRotation(newRotation);
+
+      return;
+    }
+  };
+
+  const handleLoadStart = () => {
+    console.warn('OBJ loading has started');
+  };
+  const handleLoadEnd = () => {
+    console.warn('OBJ loading has finished');
+  };
+  const handleError = (event) => {
+    console.warn('OBJ loading failed with error: ' + event.nativeEvent.error);
+  };
+
+  return (
+    <ViroARScene dragType="FixedToWorld">
+      <ViroAmbientLight color={'#fff'} />
+
+      <Viro3DObject
+        source={{ uri: object.vrx }}
+        resources={[{ uri: object.png }]}
+        type="VRX"
+        position={position}
+        scale={[0.02, 0.02, 0.02]}
+        rotation={rotation}
+        onDrag={moveObject}
+        onRotate={rotateObject}
```

Comment:
i was not able to move or rotate the construction worker on iPhone 6s. did that work properly for you?
for some reason the guy moves a bit when moving the phone. i thought the object is placed at one place and sticks to that position?

Reply:
It's working fine for me. I guess it would be a bit of a dream to expect the same performance on every device. 🤷‍♂️😔

Reply:
it was working for me with the box, but not with the worker. hm.